puppet/lib/puppet/indirector/facts/puppetdb.rb

```diff
@@ -26,7 +26,18 @@ def save(request)
         facts = request.instance.dup
         facts.values = facts.values.dup
         facts.values[:trusted] = get_trusted_info(request.node)
-        {
+
+        inventory = facts.values['_puppet_inventory_1']
+        package_inventory =
+          if inventory and inventory.respond_to?(:keys) and inventory['packages']
+            inventory['packages'].map { |p|
+              [p['package_name'], p['version'], p['provider']]
```

Comment:
not too big a deal, but seeing as this fact is controlled by us, why not just receive it as a list of tuples and skip this step?